compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -864,6 +864,17 @@ object SymDenotations {
     final def isNullableClassAfterErasure(using Context): Boolean =
       isClass && !isValueClass && !is(ModuleClass) && symbol != defn.NothingClass
 
+    /** Is `pre` the same as C.this, where C is exactly the owner of this symbol,
+     *  or, if this symbol is protected, a subclass of the owner?
+     */
+    def isCorrectThisType(pre: Type)(using Context): Boolean = pre match
```

Comment:
Now that it's available outside of `isAccessibleFrom`, the name `isCorrectThisType` is not so clear anymore. Consider renaming to `isAccessPrivilegedThisType` or something like that?